### DISCUSSION TITLE
Admin Dashboard: Changing the original default homepage

### DISCUSSION THREAD

**Discussion by m-bakoulis** [givanz/Vvveb#242] [Q&A]
Hi @givanz, I wish to change the original index of the admin dashboard homepage with something simpler like a welcome message with some simple instructions since the majority of admin users have very basic roles. Can you please guide me on how replace the original default homepage of the /admin/ ?

![Screenshot 2024-12-11 103244](https://github.com/user-attachments/assets/bf757b1e-7bd0-44fb-8e4e-1b2a0b0fec71)

Thanks in advance!

**Comment by givanz** [comment]
Hi

I made a simple plugin that loads a different page for each role, it has two html examples for `demo` and `admin` role, you can add other html files for each additional role.

With the `latest.zip` update it will work with role name.html, if you are using an older version you must rename the html files to role_id.html for example you must rename `demo.html` to `10.html` for demo role.

[admin-role-home.zip](https://github.com/user-attachments/files/18352059/admin-role-home.zip)


The plugin checks if there is a html file in the plugin folder for the current role and if the current page is homepage swaps the default template with role.html

```php
function admin() {
	Event::on('Vvveb\System\Core\View', 'compile', __CLASS__, function ($template, $htmlFile, $tplFile, $vTpl, $view) {
		
		if ($template == 'index.html') {
			$admin = Admin::current();
			if ($admin) {
				//use role name or role id which html format is available
				$role = $admin['role'] ?? $admin['role_id'] ?? 'index.html';
				$file = __DIR__ . DS . $role . '.html';
		
				//check if role html available
				if (file_exists($file)) {
					$htmlFile = $file;
				}
			}
		}

		return [$template, $htmlFile, $tplFile, $vTpl, $view];
	});
}
```

**Reply by m-bakoulis** [reply]
Sounds Awesome, I will try it.

Thanks
